### PR TITLE
Deprecate some functions related to XML digests

### PR DESCRIPTION
**Pull request by nrwahl2**

### CODE REVIEW

File: lib/common/digest.c [hunk 1]
```diff
@@ -9,11 +9,15 @@
...
 
+#include <stdbool.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <string.h>
 #include <stdlib.h>
+
+#include <glib.h>               // GString, etc.
 #include <gnutls/crypto.h>
+#include <md5.h>
```
Comment (clumens) [inline lib/common/digest.c:20]: f5b6b67c8b1d019e6c34975c9afb84823308e737 got rid of this include line, and it being back is what's causing the CI failures.

**Comment by nrwahl2** [comment]
Updated to fix CI failure caused by `md5.h` include

**Comment by kgaillot** [comment]
needs another rebase

**Comment by nrwahl2** [comment]
Rebased to resolve conflict

**Review by kgaillot — APPROVED** [review]

**Pull request merged**